source/adapters/level_zero/v2/event.hpp

```diff
@@ -0,0 +1,37 @@
+//===--------- event.hpp - Level Zero Adapter -----------------------------===//
+//
+// Copyright (C) 2024 Intel Corporation
+//
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM
+// Exceptions. See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include <stack>
+
+#include <ur/ur.hpp>
+#include <ur_api.h>
+#include <ze_api.h>
+
+#include "event_provider.hpp"
+
+namespace v2 {
+
+class ur_event {
```

Comment:
Will this be equivalent to ur_event_handle_t_?

Reply:
Yes. I don't think there's a point in reusing the existing event structure. We can just allocate this one for all events and cast it to `ur_event_handle_t *`.